colcon-build-and-test/action.yaml

```diff
@@ -51,21 +51,30 @@ runs:
         . /opt/ros/${{ inputs.rosdistro }}/setup.sh
         colcon build --event-handlers console_cohesion+ \
           --packages-above-and-dependencies ${{ inputs.target-packages }} \
-          --cmake-args -DCMAKE_BUILD_TYPE=Release -DCMAKE_EXPORT_COMPILE_COMMANDS=ON
+          --cmake-args \
+            -DCMAKE_BUILD_TYPE=Release \
+            -DCMAKE_C_FLAGS="-fprofile-arcs -ftest-coverage -DCOVERAGE_RUN=1" \
+            -DCMAKE_CXX_FLAGS="-fprofile-arcs -ftest-coverage -DCOVERAGE_RUN=1" \
+            -DCMAKE_EXPORT_COMPILE_COMMANDS=ON
       shell: bash
 
     - name: Test
       run: |
         . /opt/ros/${{ inputs.rosdistro }}/setup.sh
+        colcon lcov-result --initial
         colcon test --event-handlers console_cohesion+ \
           --packages-above ${{ inputs.target-packages }} \
           --return-code-on-test-failure
+        colcon lcov-result
+        colcon coveragepy-result
```

Comment:
Could you link some references to the commands you added here and explain what they do?

Reply:
Also, are they automatically installed by `colcon-common-extensions`?

Reply:
For `colcon lcov-result`, it collects the lcov results: 
https://github.com/colcon/colcon-lcov-result#usage

For `colcon coveragepy-result`, it collects coverage results using the coveragepy-result command:
https://github.com/colcon/colcon-coveragepy-result#usage

> Also, are they automatically installed by colcon-common-extensions?

No. I need to install these extensions via pip later.

Reply:
Thank you!

> No. I need to install these extensions via pip later.

Will you add them to this PR?

Reply:
> Will you add them to this PR?

Yes, I added 0989375